src/Board.java

```diff
@@ -97,4 +106,40 @@ private void updatePieceCount(int position) {
             this.numberOfBlackPieces--;
         }
     }
+    
+    public boolean isEndState() {
+        return (numberOfBlackPieces == 0 || numberOfWhitePieces == 0);
+    }
+
+    public List<Square> getAdjacentSquares(Square square) {
+        List<Integer> squareNumbers = square.getAdjacentSquares();
+        return getSquares((ArrayList<Integer>) squareNumbers);
+    }
+
+    public List<Square> getSquaresTwoMovesAway(Square square) {
```

Comment:
I'll change the methods to be more explicity about hops/jumps/moves, so there is no confusion in reading them.